x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -351,6 +366,8 @@ private void executeSubPlans(
         PlanRunner runner,
         EsqlExecutionInfo executionInfo,
         EsqlQueryRequest request,
+        EsqlStatement statement,
```

Comment:
I wouldn't pass the full statement down here. We can add a new parameter if needed to say whether or not we're approximating.